lib/Sema/TypeCheckDecl.cpp

```diff
@@ -4392,6 +4392,17 @@ static Type formExtensionInterfaceType(
   return resultType;
 }
 
+/// Retrieve the generic parameter depth of the extended type.
+static unsigned getExtendedTypeGenericDepth(ExtensionDecl *ext) {
+  auto nominal = ext->getSelfNominalTypeDecl();
+  if (!nominal) return static_cast<unsigned>(-1);
+
+  auto sig = nominal->getGenericSignatureOfContext();
+  if (!sig) return static_cast<unsigned>(-1);
+
+  return sig->getGenericParams().back()->getDepth();
```

Comment:
`return nominal->getGenericContextDepth();` ?

Reply:
I don't think we need a context walk here if we already have this information available in this (admittedly baroque) form.  There's quite a few places in the compiler that are computing depths this way.  Perhaps we should look into unifying them.